awx/main/models/rbac.py

```diff
@@ -733,5 +735,83 @@ def sync_parents_to_new_rbac(instance, action, model, pk_set, reverse, **kwargs)
             give_or_remove_permission(child_role, team, giving=is_giving)
 
 
+ROLE_DEFINITION_TO_ROLE_FIELD = {
+    'Organization Member': 'member_role',
+    'Controller Organization Member': 'member_role',
+    'WorkflowJobTemplate Admin': 'admin_role',
+    'Organization WorkflowJobTemplate Admin': 'workflow_admin_role',
+    'WorkflowJobTemplate Execute': 'execute_role',
+    'WorkflowJobTemplate Approve': 'approval_role',
+    'InstanceGroup Admin': 'admin_role',
+    'InstanceGroup Use': 'use_role',
+    #  'ExecutionEnvironment Admin': '',
+    'Organization ExecutionEnvironment Admin': 'execution_environment_admin_role',
+    'Project Admin': 'admin_role',
+    'Organization Project Admin': 'project_admin_role',
+    'Project Use': 'use_role',
+    'Project Update': 'update_role',
+    'JobTemplate Admin': 'admin_role',
+    'Organization JobTemplate Admin': 'job_template_admin_role',
+    'JobTemplate Execute': 'execute_role',
+    'Inventory Admin': 'admin_role',
+    'Organization Inventory Admin': 'inventory_admin_role',
+    'Inventory Use': 'use_role',
+    'Inventory Adhoc': 'adhoc_role',
+    'Inventory Update': 'update_role',
+    # 'NotificationTemplate Admin': 'admin_role',
+    'Organization NotificationTemplate Admin': 'notification_admin_role',
+    'Credential Admin': 'admin_role',
+    'Organization Credential Admin': 'credential_admin_role',
+    'Credential Use': 'use_role',
+    'Team Admin': 'admin_role',
+    'Controller Team Admin': 'admin_role',
+    'Team Member': 'member_role',
+    'Controller Team Member': 'member_role',
+    'Organization Admin': 'admin_role',
+    'Controller Organization Admin': 'admin_role',
+    'Organization Audit': 'auditor_role',
+    'Organization Execute': 'execute_role',
+    'Organization Approval': 'approval_role',
+}
+
+
+def _sync_assignments_to_old_rbac(instance, delete=True):
+    with disable_rbac_sync():
+        field_name = ROLE_DEFINITION_TO_ROLE_FIELD.get(instance.role_definition.name)
+        if not field_name:
+            return
+        try:
+            role = getattr(instance.object_role.content_object, field_name)
+        # in the case RoleUserAssignment is being cascade deleted, then
+        # object_role might not exist. In which case the object is about to be removed
+        # anyways so just return
+        except ObjectDoesNotExist:
+            return
+        if isinstance(instance.actor, get_user_model()):
+            # user
+            if delete:
+                role.members.remove(instance.actor)
+            else:
+                role.members.add(instance.actor)
+        else:
+            # team
+            if delete:
+                instance.team.member_role.children.remove(role)
+            else:
+                instance.team.member_role.children.add(role)
```

Comment:
team `member_role` vs `admin_role`?

Reply:
just `member_role`, not `admin_role`. The admin role is a parent of member role, so any parent-child relationships to other roles (like, you gave the team a permission) only needed to happen for the member role. However, what I sought to establish in https://github.com/ansible/awx/pull/15462 is that there are extremely so minimal observable API behaviors related to the old parents/children relationship, that it can already be removed. What you have is fine.